src/hooks/useReceiptValidation.js

```diff
@@ -0,0 +1,69 @@
+import {useState, useCallback} from 'react';
+import {Alert, InteractionManager} from 'react-native';
+import lodashGet from 'lodash/get';
+import _ from 'underscore';
+import useLocalize from './useLocalize';
+import * as FileUtils from '../libs/fileDownload/FileUtils';
+import CONST from '../CONST';
+
+export default function useReceiptValidation() {
+    const {translate} = useLocalize();
+    const [receiptValidation, setReceiptValidation] = useState({
+        isReceiptInvalid: false,
+        title: '',
+        reason: '',
+    });
+
+    const resetValidation = () => {
+        setReceiptValidation({
+            ...receiptValidation,
+            isReceiptInvalid: false,
+        });
+    };
+
+    const showImageCorruptionAlert = useCallback(() => {
+        Alert.alert(translate('attachmentPicker.attachmentError'), translate('attachmentPicker.errorWhileSelectingCorruptedImage'));
+    }, [translate]);
+
+    const setUploadReceiptError = useCallback((isInvalid, title, reason) => {
+        InteractionManager.runAfterInteractions(() => {
```

Comment:
On IOS, sometimes the transition is not smooth, I believe the Root cause is that closing of photos gallery and Confirm modal overlaps, so i used `InteractionManager.runAfterInteractions(() => {`, this is also reproducable while sending image in a chat. 

Note: Sometimes animation is still not smooth, The alternative solution is to use `setTimeOut` here but i have seen we avoid using it.